### PR TITLE
0.3.5 social elasticsearch

### PR DESCRIPTION
**Pull request by rwhite27**

### CODE REVIEW

File: src/Social/Contract/Channels/ChannelsTrait.php [hunk 1]
```diff
@@ -1,25 +1,24 @@
 <?php
 
-declare(strict_types=1);
+namespace Kanvas\Packages\Social\Contract\Channels;
```
Comment (kaioken) [inline src/Social/Contract/Channels/ChannelsTrait.php:4]: @rwhite27 leave the strict

**Review by kaioken — CHANGES_REQUESTED** [review]

File: src/Social/Contract/Channels/ChannelsTrait.php [hunk 2]
```diff
@@ -39,14 +38,14 @@ public static function getChannelByName(string $channelName): ChannelsModel
      * @param string $description
      * @return ChannelsModel
      */
-    public static function create(UserInterface $user, ChannelsInterface $channelEntity, string $name, string $description = ''): ChannelsModel
+    public function createChannel(UserInterface $user, string $name, string $description = ''): ChannelsModel
```
Comment (kaioken) [inline src/Social/Contract/Channels/ChannelsTrait.php:42]: @rwhite27 does description accept null? or just empy

File: src/Social/Contract/Comments/CommentsTrait.php [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+    /**
+     * Get comment by its id
+     *
+     * @param int $commentId
+     * @return Response
+     */
+    public function getComment(int $commentId): Response
+    {
+        return $this->response(Comments::getById((string)$commentId));
```
Comment (kaioken) [inline src/Social/Contract/Comments/CommentsTrait.php:45]: @rwhite27 dont we have to check that we have access to this comment by checking company and the other validations?

File: src/Social/Contract/Comments/CommentsTrait.php [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    /**
+     * Add a new comment to a message
+     *
+     * @param int $messageId
+     * @return Response
+     */
+    public function addComment(int $messageId): Response
+    {
+        $request = $this->processInput($this->request->getPostData());
+        $newComment = Comments::add($messageId, $request['message']);
```
Comment (kaioken) [inline src/Social/Contract/Comments/CommentsTrait.php]: @rwhite27 on before input we need to force when creating the user id, company id and app id

**Review by kaioken — CHANGES_REQUESTED** [review]

File: src/Social/Contract/Channels/ChannelsTrait.php (hunk 2 above)
Comment (rwhite27) [reply src/Social/Contract/Channels/ChannelsTrait.php:42]: cannot accept null

File: src/Social/Contract/Comments/CommentsTrait.php (hunk 3 above)
Comment (rwhite27) [reply src/Social/Contract/Comments/CommentsTrait.php:45]: true true. Will add those validations

File: src/Social/Contract/Comments/CommentsTrait.php (hunk 4 above)
Comment (rwhite27) [reply src/Social/Contract/Comments/CommentsTrait.php]: we already do, those are set in the comments function which is called inside the add function: 

```
public function comment(string $message): MessageComments
    {
        $comment = new MessageComments();
        $comment->message_id = $this->getId();
        $comment->apps_id = Di::getDefault()->get('app')->getId();
        $comment->companies_id = Di::getDefault()->get('userData')->getDefaultCompany()->getId();
        $comment->users_id = Di::getDefault()->get('userData')->getId();
        $comment->message = $message;
        $comment->saveOrFail();

        return $comment;
    }
```

Comment (kaioken) [reply src/Social/Contract/Comments/CommentsTrait.php]: I would change this to provide the user as a param, that way I can use that function in other places not only relying on the DI

File: src/Social/Models/SystemModules.php [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    /**
+     * Get System Module by its model_name.
+     *
+     * @deprecated v2
+     *
+     * @param string $model_name
+     *
+     * @return ModelInterface
+     */
+    public static function getSystemModuleByModelName(string $modelName) : ModelInterface
```
Comment (kaioken) [inline src/Social/Models/SystemModules.php]: @rwhite27 doesnt kanvas core have this already?

File: src/Social/Providers/ConfigProvider.php [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    public function register(DiInterface $container) : void
+    {
+        $container->setShared(
+            'config',
+            function () {
+
+                /**
+                 * @todo Find a better way to handle unit test file include
+                 */
+                $data = require appPath('vendor/canvas/core/src/Core/config.php');
```
Comment (kaioken) [inline src/Social/Providers/ConfigProvider.php]: @rwhite27 kanvs core doesn't hit have this?

File: src/Social/Tasks/ElasticTask.php [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+use Phalcon\Di;
+
+class ElasticTask extends KanvasElasticTask
+{
+    /**
+     * Handle the index of the document.
+     *
+     * @return void
+     */
+    public function indexDocumentAction()
```
Comment (kaioken) [inline src/Social/Tasks/ElasticTask.php]: @rwhite27 index feeds or messages

**Review by kaioken — CHANGES_REQUESTED** [review]

File: src/Social/Models/SystemModules.php (hunk 5 above)
Comment (rwhite27) [reply src/Social/Models/SystemModules.php]: Had to override because of the cache element.

File: src/Social/Providers/ConfigProvider.php (hunk 6 above)
Comment (rwhite27) [reply src/Social/Providers/ConfigProvider.php]: file no longer necessary, will remove it.

File: src/Social/Tasks/ElasticTask.php (hunk 7 above)
Comment (rwhite27) [reply src/Social/Tasks/ElasticTask.php]: Will change it to indexMessages

File: src/Social/Models/SystemModules.php [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
```
Comment (kaioken) [inline src/Social/Models/SystemModules.php]: @rwhite27 since you update kanvas is this needed?

File: src/Social/Services/Messages.php [hunk 9]
```diff
@@ -44,6 +45,34 @@ public static function getMessageByUuid(string $uuid): MessagesInterface
...
+
+    /**
+     * Get all the messages of a channel
+     *
+     * @param Channels $user
+     * @param array $filter
+     *
+     * @return Simple
+     */
+    public static function getByChannel(Channels $channel, int $page = 1, int $limit = 25, string $orderBy = "id", string $sort = "DESC", string $messageTypeId = null): Simple
```
Comment (kaioken) [inline src/Social/Services/Messages.php]: @rwhite27 shouldn it be ?string $messageTypeId = null

File: src/Social/Tasks/ElasticTask.php [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+        // $messagesRecords = MessagesModel::findOrFail(['limit' => 1]);
+
+        foreach ($messagesRecords as $message) {
+            $messages->setData($message->id, []);
+
+            if (!Indices::exist($messages->getIndices())) {
+                Indices::create($messages, 3, 1000);
+            }
+            $result = $messages->add();
+            Di::getDefault()->get('log')->info('Vehicle Add to Index BrowseVehicle', [$result]);
```
Comment (kaioken) [inline src/Social/Tasks/ElasticTask.php]: @rwhite27 vehicles?

File: src/Social/Contract/Comments/CommentsTrait.php (hunk 4 above)
Comment (kaioken) [reply src/Social/Contract/Comments/CommentsTrait.php]: @rwhite27 we still need to pass wthe user right?

File: src/Social/Contract/Comments/CommentsTrait.php [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+     * Add a new comment to a message
+     *
+     * @param int $messageId
+     * @return Response
+     */
+    public function editComment(int $commentId): Response
+    {
+        $request = $this->request->getPutData();
+
+        $validation = new CanvasValidation();
```
Comment (kaioken) [inline src/Social/Contract/Comments/CommentsTrait.php]: @rwhite27 d we need this validation since you have it in processinput?

**Review by kaioken — CHANGES_REQUESTED** [review]

File: src/Social/Services/Messages.php (hunk 9 above)
Comment (rwhite27) [reply src/Social/Services/Messages.php]: that function is no longer necessary for this  version. Will tag it as deprecated.

File: src/Social/Tasks/ElasticTask.php (hunk 10 above)
Comment (rwhite27) [reply src/Social/Tasks/ElasticTask.php]: lol my bad

**Pull request merged**